stl/inc/algorithm

```diff
@@ -3993,7 +3993,9 @@ namespace ranges {
 #endif // __cpp_lib_is_constant_evaluated
                 {
                     const auto _Distance = static_cast<size_t>(_ULast - _UFirst);
-                    _CSTD memset(_UFirst, static_cast<unsigned char>(_Value), _Distance);
+                    using _DestTy        = _Iter_value_t<decltype(_UFirst)>;
+                    // first cast _Value to _DestTy in case _DestTy is bool

```

Comment:
Given that we repeat this a lot I would suggest to create a helper function similar to `_Copy_memmove` maybe `_Fill_memset`?

Reply:
Good idea!

Reply:
Done